dts/bindings/display/backlight-pwm.yaml

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2024, Maritn Kiepfer
+# SPDX-License-Identifier: Apache-2.0
+
+description: Backlight PWM node
+
+compatible: "backlight-pwm"
+
+include: [base.yaml, backlight.yaml]
```

Comment:
I think the backlight.yaml could heavily borrow the properties from linux
https://www.kernel.org/doc/Documentation/devicetree/bindings/leds/backlight/common.yaml

This also allows for things like stepped brightness levels.